src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -27,16 +27,63 @@
 #![allow(clippy::mutex_atomic)]
 
 use parking_lot::Mutex;
+use rand::thread_rng;
+use rand::Rng;
+use futures::task_local;
+use std::sync::Arc;
 
 pub struct WorkUnit {
   pub name: String,
   pub start_timestamp: f64,
   pub end_timestamp: f64,
   pub span_id: String,
+  pub parent_id: Option<String>,
 }
 
-pub trait WorkUnitStore {
-  fn should_record_zipkin_spans(&self) -> bool;
-  fn get_workunits(&self) -> &Mutex<Vec<WorkUnit>>;
-  fn add_workunit(&self, workunit: WorkUnit);
+#[derive (Clone)]
+pub struct WorkUnitStore {
+  workunits: Arc<Mutex<Vec<WorkUnit>>>,
+}
+
+impl WorkUnitStore {
+  pub fn new() -> WorkUnitStore {
+    WorkUnitStore {
+      workunits: Arc::new(Mutex::new(Vec::new())),
+    }
+  }
+
+  pub fn get_workunits(&self) -> Arc<Mutex<Vec<WorkUnit>>> {
+    self.workunits.clone()
+  }
+
+  pub fn add_workunit(&self, workunit: WorkUnit) {
+    self.workunits.lock().push(workunit);
+  }
+
+  pub fn len(&self) -> usize {
```

Comment:
`len` seems like a kind of weird operation to have on a `WorkUnitStore`, and it looks like it's only used in a test. Maybe in the test instead do `store.get_workunits().lock().len()`?